### PR TITLE
Feature/plan detail

### PR DESCRIPTION
**Pull request by doctype-Melvin**
PR for [US2](https://github.com/doctype-Melvin/capstone/issues/3)

- adds database folder containing ODM
- adds single page view page 
- adds form submit handler function and swr hook
- changes folder names and adds API endpoints
- adds basic styling to WorkoutDay component and SinglePlanView

### CODE REVIEW

File: components/workoutDay/WorkoutDay.js [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    box-shadow: 1px 1px 2px 2px rgba(0, 0, 0, 0.1);
+    max-width: fit-content;
+    padding: .3rem;
+    border-radius: 3px;
+`
+
+const StyledTitel = styled.h1`
```
Comment (NikRoe) [inline components/workoutDay/WorkoutDay.js]:
```suggestion
const StyledTitle = styled.h1`
```

File: components/workoutDay/WorkoutDay.js [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+    fontSize: 1rem;
+    max-width: fit-content;
+`
+
+export default function WorkoutDay({number}){
+    return (
+        <StyledSection>
+            <StyledTitel>
+                Day {number}
+            </StyledTitel>
```
Comment (NikRoe) [inline components/workoutDay/WorkoutDay.js]:
```suggestion
            <StyledTitle>
                Day {number}
            </StyledTitle>
```

File: components/workoutDay/WorkoutDay.js [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+            <StyledButton>
```
Comment (NikRoe) [inline components/workoutDay/WorkoutDay.js]: try to be explicit about the types

```suggestion
            <StyledButton type="button">
```

File: pages/api/plans/[id].js [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+    await dbConnect()
+
+    const { id } = request.query
+
+    if (request.method === "GET") {
+        const plan = await Plan.findById(id)
+        if (!plan) {
+            return response.status(404).json({status: "Plan not found"})
+        }
+        response.status(200).json(plan)
```
Comment (NikRoe) [inline pages/api/plans/[id].js]: you can consider wrapping this in a try-catch-block

File: pages/viewPlans/[id].js [hunk 5]
```diff
@@ -0,0 +1,54 @@
...
+const sharedStyleRules = `
+    width: 100vw;
+`
+
+const PlanContainer = styled.section`
+    border: 1px solid red;
+    height: 100vh;
+`
+
+const PlanHead = styled.section`
```
Comment (NikRoe) [inline pages/viewPlans/[id].js]: a paragraph makes more sense here:

```suggestion
const PlanHead = styled.p`
```

File: pages/viewPlans/[id].js [hunk 6]
```diff
@@ -0,0 +1,54 @@
+import styled from "styled-components"
+import useSWR from "swr"
+import { useRouter } from "next/router"
+import Link from "next/link"
```
Comment (NikRoe) [inline pages/viewPlans/[id].js]: you don't use the link component here

File: pages/viewPlans/index.js [hunk 7]
```diff
@@ -0,0 +1,7 @@
+export default function ViewAllPlans() {
+    return (
+        <>
+        <div>Here Will Be All Plans</div>
+        </>
+    )
```
Comment (NikRoe) [inline pages/viewPlans/index.js]: you don't need fragments:

```suggestion
    return <div>Here Will Be All Plans</div>
```

**Review by NikRoe — CHANGES_REQUESTED** [review]
Hi Melvin,

this looks really great and impressive so far!

I left only a few notes for you to look at.

Keep it up!

**Comment by doctype-Melvin** [comment]
Thanks. Really appreciate your feedback.

File: components/workoutDay/WorkoutDay.js [hunk 8]
```diff
@@ -0,0 +1,36 @@
+import styled from "styled-components"
+
+const StyledSection = styled.section`
+    display: flex;
+    flex-direction: column;
+    align-items: center;
+    box-shadow: 1px 1px 2px 2px rgba(0, 0, 0, 0.1);
+    max-width: fit-content;
```
Comment (CJAkkuec) [inline components/workoutDay/WorkoutDay.js:8]:
```suggestion
    width: fit-content;
```

Comment (CJAkkuec) [reply components/workoutDay/WorkoutDay.js:8]: In this instance, you don't need the "max-", because either property leads to the same result.

**Review by CJAkkuec — CHANGES_REQUESTED** [review]
Just a small note.

**Comment by mbosselmann** [comment]
LGTM! 🦊

**Comment by doctype-Melvin** [comment]
Thanks. This time I didn't forget about the QA.

**Pull request merged**